src/components/screens/question-form.tsx

```diff
@@ -0,0 +1,170 @@
+import {
+  Box,
+  Button,
+  Card,
+  CardActionArea,
+  CardActions,
+  CardContent,
+  Checkbox,
+  TextField,
+  Typography
+} from "@mui/material";
+import type React from "react";
+import { useState } from "react";
+
+/**
+ * NewQuestionCard component
+ * Interface for the NewQuestionCard component
+ */
+
+interface NewQuestionProps {
+  handleAddQuestionSubmit: (
+    questionText: string,
+    options: { label: string; value: boolean }[]
+  ) => void;
+}
+
+const NewQuestionCard: React.FC<NewQuestionProps> = ({ handleAddQuestionSubmit }) => {
+  /**
+   * State for the question text and options
+   */
+  const [questionText, setQuestionText] = useState("");
+  const [options, setOptions] = useState([
+    { label: "", value: false },
+    { label: "", value: false },
+    { label: "", value: false },
+    { label: "", value: false }
+  ]);
+
+  /**
+   * Handle options label change
+   */
+  const handleAnswerLabelChange = (
+    index: number,
+    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+  ) => {
+    const updatedOptions = [...options];
+    updatedOptions[index].label = event.target.value;
+    setOptions(updatedOptions);
+  };
+
+  /**
+   * Handle options value change (checkbox)
+   * @param index
+   */
+  const handleCheckboxChange = (index: number) => {
+    const updatedOptions = [...options];
+    updatedOptions[index].value = !updatedOptions[index].value;
+    setOptions(updatedOptions);
+  };
+
+  /**
+   * Handle adding a new option (adds an empty option to the list)
+   */
+  const handleAddNewOption = () => {
+    setOptions([...options, { label: "", value: false }]);
+  };
+
+  /**
+   * Handle saving the question (submitting the question and options + resetting the form)
+   */
+  const handleSaveQuestion = () => {
+    handleAddQuestionSubmit(questionText, options);
+
+    setQuestionText("");
+    setOptions([
+      { label: "", value: false },
+      { label: "", value: false },
+      { label: "", value: false },
+      { label: "", value: false }
```

Comment:
Update this to match the imporvement suggested for the initial setting of the state.